### ISSUE TITLE
[Bug] Engine space should be ignore the version change

### ISSUE THREAD

**Issue by ulysses-you** [apache/incubator-kyuubi#2734]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the bug

We use `${serverSpace}_${KYUUBI_VERSION}_${shareLevel}_${engineType}` as the parent path of the engine space, but during hot upgrade kyuubi cluster with different version, we will have two version of engine space. As a result, a user share level engine can be created again since there are two engine space at that monent.

```
Client A -> Kyuubi Instance 1.5.0 -> engine A
Client A -> Kyuubi Instance 1.5.1 -> engine B
```

### Affects Version(s)

master/1.5

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by ulysses-you** [comment]
I plan to change `KYUUBI_VERSION` to a static version number e.g. 1. But it affects the exists kyuubi ctl. what do you think @yaooqinn @pan3793

**Comment by yaooqinn** [comment]
versions are used to prevent breaking change，but for patch release，it seems overkilling

**Comment by ulysses-you** [comment]
yeah, one option is we use a internal version for it like the protocol version of thrift. It is not coupled with kyuubi version, but we can change it once kyuubi version has breaking change.

**Comment by ulysses-you** [comment]
We can probably handle this issue at client side. e.g. At jdbc client, we always choose a bigger version kyuubi server to connect.

**Comment by cxzl25** [comment]
> Yes, replace KYUUBI_VERSION by something like PROTOCOL_VERSION is more reasonable to me, and we should only bump PROTOCOL_VERSION when breaking change happens

+1
Using a custom protol version looks fine.

Upgrading from 1.4 to 1.5 today, I had to restart N engines.
Similar to manually killing the old ones, because the idle time is set relatively high.
